django/db/models/fields/related_descriptors.py

```diff
@@ -573,6 +573,12 @@ def _apply_rel_filters(self, queryset):
             queryset._known_related_objects = {self.field: {self.instance.pk: self.instance}}
             return queryset
 
+        def _remove_prefetched_objects(self):
+            try:
+                self.instance._prefetched_objects_cache.pop(self.field.related_query_name())
+            except (AttributeError, KeyError):
```

Comment:
You should add new unit tests that fail if `KeyError` is removed from the new code.